backend/inspirehep/access_control.py

```diff
@@ -43,3 +43,13 @@ def can(self):
 
 def session_cataloger_permission_factory(*args, **kwargs):
     return SessionCatalogerPermission()
+
+
+class LiteraturePermissionCheck:
+    def __init__(self, record, *args, **kwargs):
+        self.record = record
+
+    def can(self):
+        if "Literature" not in self.record.get("_collections", []):
+            return SessionCatalogerPermission().can()
+        return True
```

Comment:
why we did it like a class? I'd follow the invenio docs just to be aligned with them https://github.com/inveniosoftware/invenio-records-rest/blob/69c8d45361e898a2689f7c1442395261f701dfbb/invenio_records_rest/config.py#L71

Reply:
It looks like class, it's called like class: https://labs.inspirehep.net/literature/1767665 But in Invenio they decided that it's better to create small ugly thing... Both are callables, and it will work both with function and class but class here is much more clear.

Like for real... Is this more readable in your opinion:
```python
def permission_check_factory():
        def check_permission(record, *args, **kwargs):
            def can(self):
                if "Literature" not in self.record.get("_collections", []):
                    return SessionCatalogerPermission().can()
            return type('Check', (), {'can': can})()
```
than this
```python
class LiteraturePermissionCheck:
    def __init__(self, record, *args, **kwargs):
        self.record = record

    def can(self):
        if "Literature" not in self.record.get("_collections", []):
            return SessionCatalogerPermission().can()
        return True
```

Reply:
so instead of having a static class that is configurable enough for their needs, they have a function that
1. builds a function to do the permission check
2. builds a dynamic class with that function as a method and instantiate it
:man_facepalming:

Reply:
Yup... probably for "better code quality"...

Reply:
"hey, dynamic classes are cool.. so let's use them!" 😄 

btw I did `SessionCatalogerPermission` with a proper class, exactly for this reason because the other way is retarded (at least for our usecase)